### PR TITLE
Make cudf::test::expect_columns_equal() to fail when comparing unsanitary lists.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Fixes:  https://github.com/rapidsai/cudf/issues/10855

Unsanitary lists (those with null rows that actually have backing underlying data) should not compare as equal to sanitized lists, but _should_ compare as equivalent.   Only one test in cudf was affected by this change, which I've opened a second issue for:
https://github.com/rapidsai/cudf/issues/10856

This PR also adds a utility function to `cudf::test::lists_column_wrapper`, 

`static lists_column_wrapper<T> make_one_empty_row_column(bool valid = true)`

Due to the way initializer lists and nested class initialization work, it was impossible to create a list column with 1 row that was empty, so I added this explicit function to handle the edge case.

### CODE REVIEW

File: cpp/tests/quantiles/percentile_approx_test.cu [hunk 1]
```diff
@@ -404,7 +404,8 @@ TEST_F(PercentileApproxTest, EmptyInput)
                             3,
                             cudf::test::detail::make_null_mask(nulls.begin(), nulls.end()));
 
-  CUDF_TEST_EXPECT_COLUMNS_EQUAL(*result, *expected);
+  // TODO: change percentile_approx to produce sanitary list outputs for this case.
```
Comment (vuule) [inline cpp/tests/quantiles/percentile_approx_test.cu:407]: is this for a future PR?

File: cpp/include/cudf_test/column_wrapper.hpp [hunk 2]
```diff
@@ -1441,7 +1441,44 @@ class lists_column_wrapper : public detail::column_wrapper {
...
+  static lists_column_wrapper<T> make_one_empty_row_column(bool valid = true)
+  {
+    cudf::test::fixed_width_column_wrapper<cudf::offset_type> offsets{0, 0};
+    cudf::test::fixed_width_column_wrapper<int> values{};
+    return lists_column_wrapper<T>(
+      1,
+      offsets.release(),
+      values.release(),
+      !valid ? 1 : 0,
+      !valid ? cudf::create_null_mask(1, cudf::mask_state::ALL_NULL) : rmm::device_buffer{});
```
Comment (ttnghia) [inline cpp/include/cudf_test/column_wrapper.hpp:1458]: Just `static_cast<int>(!valid)`.

File: cpp/tests/quantiles/percentile_approx_test.cu (hunk 1 above)
Comment (nvdbaranec) [reply cpp/tests/quantiles/percentile_approx_test.cu:407]: Yes.   Issue for it is here: https://github.com/rapidsai/cudf/issues/10856

**Review by ttnghia — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

**Comment by nvdbaranec** [comment]
@gpucibot merge

**Pull request merged**